src/DocumenterTools.jl

```diff
@@ -95,4 +106,46 @@ function generate(pkg::Module; dir=nothing)
     nothing
 end
 
+"""
+    DocumenterTools.generate(pkg::Module; dir = "docs")
+
+Same as `generate(path::String)` but the `path` and name is determined
+automatically from the module.
+
+!!! note
+    The package must be in development mode. Make sure you run
+    `pkg> develop pkg` from the Pkg REPL, or `Pkg.develop(\"pkg\")`
+    before generating docs.
+
+# Examples
+```julia-repl
+julia> using DocumenterTools
+
+julia> using MyPackage
+
+julia> DocumenterTools.generate(MyPackage)
+[ ... output ... ]
+```
+"""
+function generate(pkg::Module; dir = "docs")
+    name = String(nameof(pkg))
+    path = pathof(pkg)
```

Comment:
I was thinking that it might be worth having a `parentmodule(pkg) == pkg` check here, to make sure the user called the function with a package module, not with some submodule.

Reply:
Good idea!